File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 1;
    
    let idx = 0;
    let len = priorities.length;
    
    let visited = [];
    for(let i=0; i<len ; i++) visited.push(false);
    
    while (answer <= len) {
        let max = -1;
        
        const tmp = idx;
        
        for(let i = tmp; i < tmp + len; i++) {
            const loc = i % len;
            
            if(!visited[loc] &&priorities[loc] > max) {
                    idx = loc;
                    max = priorities[loc];
            }
        }
        
        if(idx === location) break;
        visited[idx] = true;
        answer +=1;
        idx = (idx +1) % len;
        
        console.log(answer, visited, idx, max);
    }
    
    return answer;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <vector>

using namespace std;


int solution(vector<int> priorities, int location) {
    int answer = 1;
    
    int len = priorities.size();
    int idx = 0;
    vector<bool> visited(len);
    
    while(answer < len) 
    {
        int max = -1;
        int tmp = idx;
        
        for(int i=tmp; i<tmp + len; i++)
        {
            int loc = i %len;
            
            if(!visited[loc] && priorities[loc] > max) 
            {
                idx = loc;
                max = priorities[loc];
            }
        }
        
        if(idx == location) break;
        answer +=1;
        visited[idx] = true;
    }
    
    
    return answer;
}